NethermindNode.Tests/Tests/Receipts/ReceiptsVerification.cs

```diff
@@ -0,0 +1,226 @@
+  private const string WsAddress = "ws://localhost:8545";
+  private Web3 w3 = new Web3(RpcAddress);
+  private static readonly NLog.Logger Logger = NLog.LogManager.GetLogger(TestContext.CurrentContext.Test.Name);
+  public Queue<Block> blocks = new Queue<Block>();
+
+  [Test]
+  [Category("ReceiptsNew")]
+  public async Task Verify_New_Receipts()
+  {
+    var testRunTime = 2 * 60 * 1000; // 10 minutes
+
+    var client = new StreamingWebSocketClient(WsAddress);
+    var subscription = new EthNewBlockHeadersSubscription(client);
+
+    // attach our handler for new block header data
+    subscription.SubscriptionDataResponse += SubscriptionHandler;
+
+    bool subscribed = true;
+
+    // handle unsubscription
+    // optional - but may be important depending on your use case
+    subscription.UnsubscribeResponse += (object sender, StreamingEventArgs<bool> success) =>
+    {
+      subscribed = false;
+      Logger.Info($"Unsubscribed: {success.Response}");
+    };
+
+    // open the web socket connection
+    await client.StartAsync();
+
+    // subscribe to new block headers
+    // blocks will be received on another thread
+    // therefore this doesn't block the current thread
+    await subscription.SubscribeAsync();
+
+
+    Logger.Info("Waiting for new blocks: ");
+
+    var sw = new Stopwatch();
+    sw.Start();
+    //allow time to unsubscribe
+    while (subscribed)
+    {
+      if (blocks.Count > 0)
+      {
+        var block = blocks.Dequeue();
+        Logger.Info($"Processing: {block.BlockHash}");
+        /*
+        1. On synced node subscribe to new blocks using eth_subscribe with newHeads topic
+        2. Whenever notification from subscription is received we should:
+            1. Get ReceiptsRoot from new block
+            2. Get new block hash
+            3. Use block hash and call **eth_getBlockReceipts** and calculate hash of all receipts (use Nethermind method for that or craft a new one to have a separate implementation).
+            4. Compare Hashes from point i and iii.
+        3. Thing to consider: we are “losing receipts” from time to time in unknown way so maybe such tool should not only check once after subscription notification but “recheck” also after some time (maybe after pruning interval?) to make sure receipts are properly flushed to DB
+        */
+
+        var receipts = w3.Eth.Blocks.GetBlockReceiptsByNumber.SendRequestAsync(new HexBigInteger(block.Number.Value)).Result;
+        var calculatedRoot = ReceiptsHelper.CalculateRoot(receipts);
+        var receiptsRoot = block.ReceiptsRoot;
+        Logger.Info($"ReceiptsRoot: {receiptsRoot} CalculatedRoot: {calculatedRoot} {calculatedRoot == receiptsRoot}");
+        Assert.That(calculatedRoot, Is.EqualTo(receiptsRoot));
+      }
+      if (sw.ElapsedMilliseconds > testRunTime)
+      {
+        await subscription.UnsubscribeAsync();
+        subscribed = false;
+        await client.StopAsync();
+        break;
+      }
+    }
+  }
```

Comment:
Should we at the end make a re-verification of the set? We have all ReceiptsRoots calculated, we could potentially do a loop once more for each block from a set, check its receipts and make sure that receipts root did not changed (so there is no issue with flushing to disk).